### PR TITLE
Initial deployment setup

### PR DESCRIPTION
**Pull request by elgatovital**
Prepare project for deployment.

Changes
- Add nginx template
- Add docker image
- Add deployment step to ci workflow
- Add docker compose file for local deployment

TODO:
- configure env variables for staging and prod env
- add separate github workflows for prod & staging

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          fetch-depth: '0'
+      - name: Set Version
+        run: |
+          description="$(git describe --tags --always)"
+          echo "VERSION=${description:1}" >> $GITHUB_ENV
```
Comment (fmrsabino) [inline .github/workflows/deployment.yml:19]: Why do we need to set the version? This branch is triggered on `main`.

Also we don't use the `VERSION` environment variable at all in the service 🤔

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+name: safe-gelato-relay-service
+
+on: [ push, pull_request ]
+
+jobs:
+
+  docker-publish-staging:
```
Comment (fmrsabino) [inline .github/workflows/deployment.yml:7]: We have deployment jobs in both `.github/workflows/ci.yml` and `.github/workflows/deployment.yml` that target the same environment.

Why is that? 🤔

File: .github/workflows/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        with:
+          context: .
+          push: true
+          build-args: VERSION=${{ env.VERSION }}
+          tags: safeglobal/safe-client-gateway:staging
+          cache-from: type=registry,ref=safeglobal/safe-client-gateway:staging
+          cache-to: type=inline
+
+  deploy:
+    runs-on: ubuntu-20.04
```
Comment (fmrsabino) [inline .github/workflows/deployment.yml:38]: We can use the latest version available or is there any constraint regarding this?

File: .github/workflows/deployment.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        run: bash scripts/deploy_docker.sh develop
+
+  autodeploy:
+    runs-on: ubuntu-20.04
+    needs: [docker-publish-staging]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Deploy staging
+      run: bash scripts/autodeploy.sh
```
Comment (fmrsabino) [inline .github/workflows/deployment.yml:75]: We don't have this file. This is matches the CI pipeline file from the CGW. Why are we reusing all the steps from the CGW (Rust)?

File: scripts/deploy_docker.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (fmrsabino) [inline scripts/deploy_docker.sh:1]: Why do we need this file?

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+version: '3.5'
+
+services:
+
+  web:
+    build: .
+    tty: true
+    environment:
+      EXCHANGE_API_KEY: ${EXCHANGE_API_KEY-example_api_key}
```
Comment (fmrsabino) [inline docker-compose.yml:9]: We don't have an `EXCHANGE_API_KEY` in the service.

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: .github/workflows/deployment.yml (hunk 2 above)
Comment (elgatovital) [reply .github/workflows/deployment.yml]: removed deployment.yml

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -37,3 +37,26 @@ jobs:
       - run: yarn install --immutable
       - run: yarn run build
       - run: yarn run test
+
+  docker-publish:
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
+    needs: [ prettier, es-lint, tests-finish ]
```
Comment (fmrsabino) [inline .github/workflows/ci.yml]: We don't have `tests-finish` in the CI

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -37,3 +37,26 @@ jobs:
...
+        with:
+          platforms: arm64
+      - uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - uses: docker/build-push-action@v3
+        env:
+          DOCKER_IMAGE_TAG: safeglobal/safe-gelato-relay-service:staging
```
Comment (fmrsabino) [inline .github/workflows/ci.yml:57]: Why did you extract this variable? It's only used in one place. I  don't think we should set it in the `env`

File: nginx/templates/nginx.conf.template [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+http {
+  include mime.types;
+  # fallback in case we can't determine a type
+  default_type application/octet-stream;
+  sendfile on;
+
+  upstream app_server {
+    ip_hash;  # For load-balancing
+    server web:${ROCKET_PORT} fail_timeout=0;
```
Comment (fmrsabino) [inline nginx/templates/nginx.conf.template:17]: Where do you get `ROCKET_PORT` from?

**Review by fmrsabino — CHANGES_REQUESTED** [review]

**Pull request merged**